CoreImportExportPlugins/src/au/gov/asd/tac/constellation/plugins/importexport/TemplateUtilities.java

```diff
@@ -0,0 +1,167 @@
+        int tabCount = 0;
+        final JsonNode source = TemplateUtilities.getRequiredFieldFromTemplate(root, SOURCE);
+        final String destination = TemplateUtilities.getRequiredFieldFromTemplate(source, DESTINATION).textValue();
+        final SchemaFactory schemaFactory = SchemaFactoryUtilities.getSchemaFactory(destination);
+        final boolean schemaInit = TemplateUtilities.getRequiredFieldFromTemplate(source, SCHEMA_INIT).booleanValue();
+
+        importController.setDestination(new SchemaDestination(schemaFactory));
+        importController.setSchemaInitialised(schemaInit);
+        final boolean showAllSchemaAttributes = source.get(SHOW_ALL_SCHEMA_ATTRIBUTES) != null
+                && source.get(SHOW_ALL_SCHEMA_ATTRIBUTES).booleanValue();
+
+        ((ImportController) importController).getImportPane().setTemplateOptions(showAllSchemaAttributes);
+
+        final ArrayNode definitionsArray = (ArrayNode) root.withArray(DEFINITIONS);
+        for (final JsonNode definitionNode : definitionsArray) {
+            tabCount++;
+            final int firstRow = TemplateUtilities.getRequiredFieldFromTemplate(definitionNode, FIRST_ROW).intValue();
+
+            final RowFilter filter = populateFilter(definitionNode);
+            final ImportDefinition impdef = new ImportDefinition("", firstRow, filter);
+
+            final List<String> missingUserAttributes = populateImportDefinitions(importController, definitionNode, impdef);
+            definitions.add(impdef);
+
+            if (!missingUserAttributes.isEmpty()) {
+                final String message = String.format("The template `%s` is outdated or corrupted. In the tab `Run %d` following attributes are considered user added "
+                        + "for the destination `%s` with %s, as specified in the template. Hence they require `attribute_type` property. \n\n %s \n\n Consider using a new "
+                        + "template or add the missing attributes manually.", templName, tabCount, schemaFactory.getLabel(),
+                        (showAllSchemaAttributes ? "`showAllSchemaAttributes` enabled" : "`showAllSchemaAttributes` disabled"), Arrays.toString(missingUserAttributes.toArray()));
+
+                NotifyDisplayer.displayAlert(LOAD_TEMPLATE, "Template Error", message, Alert.AlertType.WARNING);
+            }
+        }
+        importController.setClearManuallyAdded(false);
+        return definitions;
+    }
+
+    private static RowFilter populateFilter(final JsonNode definitionNode) {
+        final RowFilter filter = new RowFilter();
+        if (definitionNode.has(FILTER)) {
+            final JsonNode filterNode = TemplateUtilities.getRequiredFieldFromTemplate(definitionNode, FILTER);
+            final String script = TemplateUtilities.getRequiredFieldFromTemplate(filterNode, SCRIPT).textValue();
+            final JsonNode columnsArray = filterNode.withArray(COLUMNS);
+            final ArrayList<String> columns = new ArrayList<>();
```

Comment:
Can you make the left hand side just a `List<String>`

Reply:
Just copy pasted, fixed now.